User MVC 계층 구조 설계

hjm12223 opened the pull request:
## 개요
#12
## PR 유형
어떤 변경 사항이 있나요?

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [x] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.  Commit message convention 참고  (Ctrl + 클릭하세요.) 
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

> ### Description

- 기본적인 MVC 패키지 구조 설계
- Entity 및 필요한 의존성 추가

> ### Core Code

```java

```

hjm12223 committed:
feat: 기본적인 MVC, 프로젝트 세팅

hjm12223 committed:
feat: 기본적인 MVC, 프로젝트 셋팅

no-oneho commented on user-api/src/main/java/org/ecommerce/userapi/entity/Address.java:
```diff
@@ -0,0 +1,59 @@
+	private LocalDateTime createDate;
+
+	@Column(name = "is_deleted")
+	private boolean isDeleted;
```
boolean 대신 Boolean같은 래핑타입으로 선언하는게 좋아보입니다!

hjm12223 replied:
말씀 주셔서 감사합니다

Joowon-Seo commented on user-api/src/main/java/org/ecommerce/userapi/entity/Users.java:
```diff
@@ -0,0 +1,69 @@
+	@Column(name = "is_deleted")
+	private boolean isDeleted;
+
+	@CreationTimestamp
+	@Column(name = "create_date")
+	private LocalDateTime createDate;
+
+	@UpdateTimestamp
+	@Column(name = "update_date")
+	private LocalDateTime updateDate;
+
```
종민님 벌써 많이 작성하셨네요!
회의를 통해서 프로젝트 날짜 관련 type에 대해서 정리해야할 것 같네요.
https://dev.gmarket.com/49 여기 참고하시면 좋을 것 같습니다!

no-oneho commented on user-api/src/main/java/org/ecommerce/userapi/entity/Address.java:
```diff
@@ -0,0 +1,59 @@
+
+
+	@CreationTimestamp
+	@Column(name = "create_date")
```
CreationTimestamp을 사용하실때 Column 어노테이션안에 updatable = false 옵션이 있어야하는것으로 알고있습니다!

no-oneho commented on user-api/src/main/java/org/ecommerce/userapi/entity/Address.java:
```diff
@@ -0,0 +1,59 @@
+	@Column(name = "post_address", nullable = false)
+	private String postAddress;
+
+	@Column(name = "detail", nullable = false)
```
테스트 코멘트 입니다. 수정부탁드립니다. 테스트입니다.

hjm12223 committed:
refactor: 코드 수정

[Enum 클래스 위치 변경, Entity -> Entity/type,
createDate-> createDateTime 으로 변경
createDateTime 컬럼에 updatable 추가]

no-oneho commented on user-api/src/main/java/org/ecommerce/userapi/entity/type/Status.java:
```diff
@@ -0,0 +1,15 @@
+
+@Getter
+public enum Status {
+		GENERAL("일반"),
+		WITHDRAWAL("탈퇴"),
+		SUSPEND("정지");
```
다른 파일과 들여쓰기 컨밴션이 조금 다른 것 같습니다!

no-oneho reviewed (approved):
고생하셨습니다!!

zmfpdl64 commented:
수고하셨습니다! 👍

hjm12223 committed:
refactor: 코드 수정

[Enum 들여쓰기 수정]
